activation/atxdb.go

```diff
@@ -617,12 +633,20 @@ func (db *DB) GetNodeLastAtxID(nodeID types.NodeID) (types.ATXID, error) {
 	if exists := nodeAtxsIterator.Last(); !exists {
 		return *types.EmptyATXID, ErrAtxNotFound(fmt.Errorf("atx for node %v does not exist", nodeID.ShortString()))
 	}
-	return types.ATXID(types.BytesToHash(nodeAtxsIterator.Value())), nil
+
+	return types.ATXID(types.BytesToHash(nodeAtxsIterator.Value())), err
 }
 
 // GetEpochAtxs returns all valid ATXs received in the epoch epochID
 func (db *DB) GetEpochAtxs(epochID types.EpochID) (atxs []types.ATXID) {
 	atxIterator := db.atxs.Find(getEpochPrefix(epochID))
+	defer func() {
+		atxIterator.Release()
+		if err := atxIterator.Error(); err != nil {
+			db.log.Panic("atxdb iterator error")
```

Comment:
Probably not great to add a panic. Why not do the same as above, and have this method just return an error in this case?

Reply:
hey @lrettig - this is the last task I'm working on and changing the return types causes a lot of changes around the codebase. i've to update interfaces/change values in blockoracle.go, node.go and others. Just to check this is planned/intentional?

Reply:
Thanks for checking. I just noticed that there's already a panic statement below in this method, so I think it's not so terrible to add another one. If we feel that this method signature should be changed to return an error, and that these panics should be removed, we can open a new issue to follow up on that.